### PR TITLE
VehicleInfo interface: SDL behavior in case HMI does not respond to IsReady request or respond with "available" = false

### PR DESCRIPTION
**Pull request by dtrunov**
Added logic for processing case when HMI does not respond to IsReady request or respond with
"available"=false.
 Based on implementation CRQ [APPLINK-20920](https://adc.luxoft.com/jira/browse/APPLINK-20920)
 
CRQ [APPLINK-25201](https://adc.luxoft.com/jira/browse/APPLINK-25201)

### CODE REVIEW

**Comment by dtrunov** [comment]
@LuxoftAKutsan  @AGaliuzov @AGritsevich @wolfylambova22   please review

**Comment by AGritsevich** [comment]
@dtrunov , Reviewed.

File: src/components/application_manager/include/application_manager/hmi_interfaces.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+#ifndef SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_H_
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_HMI_INTERFACES_H_
+
+#include "interfaces/HMI_API.h"
+
+namespace application_manager {
+
+class HmiInterfaces {
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/hmi_interfaces.h]: @dtrunov where is doxygen comments?

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 2]
```diff
@@ -842,6 +843,10 @@ class ApplicationManagerImpl
     return resume_ctrl_;
   }
 
+  HmiInterfaces& hmi_interfaces() OVERRIDE {
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/application_manager_impl.h]: @dtrunov such interface brakes encapsulation. So please either return const reference or do no return this class at all

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 3]
```diff
@@ -194,6 +195,55 @@ bool CommandRequestImpl::CheckSyntax(const std::string& str,
   return true;
 }
 
+smart_objects::SmartObject CreateUnsupportedResourceResponse(
+    const hmi_apis::FunctionID::eType function_id,
+    const uint32_t hmi_correlation_id,
```
Comment (wolfylambova22) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @dtrunov const names better be kFunctionId,kHmiCorrelationId maybe?

Comment (dtrunov) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @wolfylambova22 As far as I know this rule doesn't apply to paramters of methods.

File: src/components/application_manager/include/application_manager/hmi_interfaces.h (hunk 1 above)
Comment (dtrunov) [inline src/components/application_manager/include/application_manager/hmi_interfaces.h]: @AGaliuzov Kutsan A. will add comments.

**Comment by dtrunov** [comment]
@LuxoftAKutsan @AGaliuzov @wolfylambova22 please finish review.

**Comment by wolfylambova22** [comment]
@dtrunov Reviewed.

**Comment by AGaliuzov** [comment]
@dtrunov  Reviewed, just don't forget to move class's documentation

**Comment by dtrunov** [comment]
@anosach-luxoft Please merge these changes

**Pull request merged**